README.md

```diff
@@ -1,116 +1,208 @@
-In case double-click does not work, install with the command 
-`java -jar s3mInstaller.jar`.
+    int calc(int a, int b) {
+<<<<<<< MINE
+        return a * b;
+=======
+        return a + b;
+>>>>>>> YOURS
+    }
 
-#### Manual installation
-1. Copy the [binary](https://github.com/guilhermejccavalcanti/jFSTMerge/blob/master/binary/jFSTMerge.jar) file to your `$HOME` folder.
-2. Add the following lines to your `.gitconfig` file (typically localized in the `$HOME` folder).
-	```
-    [core]
-		attributesfile = ~/.gitattributes
-	[merge "s3m"]
-		name = semi_structured_3_way_merge_tool_for_java
-		driver = java  -jar "\"$HOME/jFSTMerge.jar\"" -f %A %O %B -o %A -g
-    ```
-3. Add the following line to your `.gitattributes` file (also localized in the `$HOME` folder)
-	
-    `*.java merge=s3m`
-    
+    int factorial(int n) {
+        if (n == 0) {
+            return 1;
+        } else {
+            return factorial(n - 1) * n;
+        }
+    }
 
-## Usage
+}
+```
 
-#### Statistics
-- Usage data (such as the number of detected conflicts, number of merged scenarios, and more useful details for studying the benefits and drawbacks of the tool) is stored in the `$HOME/.jfstmerge` folder.
-- A summary of collected statistics that might help one decide to continue using the tool is available in the `jfstmerge.summary` file.
+---
 
-#### Running with _git_
+## Conflict Handlers
 
-After installation, the tool is automatically integrated with git, with no need for further configuration. Then every time you invoke the `git merge` command, the tool is executed.
 
-#### Running standalone
+Is semistructured merge the ultimate solution? Not so easy. Despite reducing a lot of textual merge's false positives and false negatives, it introduces others on its own! We call them **extra** false positives/negatives. To help fixing those, S3M runs special mechanisms after every merge named as **Conflict Handlers**: algorithms that fix some semistructured merge potential flaws.
 
-Use the jar from the [/binary](https://github.com/guilhermejccavalcanti/jFSTMerge/tree/master/binary) folder, or from the installed folder.
+### [Deletions Handler](https://github.com/guilhermejccavalcanti/jFSTMerge/blob/master/src/main/java/br/ufpe/cin/mergers/handlers/DeletionsHandler.java)
 
-* Merging 3 files:
+* **Description** Every declaration removal by a contribution is problematic for semistructured merge as the correspondent node doesn't exist in the contribution's tree. This handler focuses on deletions of **inner classes**.
+* **Behavior**
+    If a contribution edited an inner class while another deleted it, we check if the latter also included a reference to the edited class. If true, we keep both the original and edited classes. Otherwise, we check if they're "sufficiently similar". If true, we join them to avoid duplications, otherwise, we report a conflict.
```

Comment:
confuso. tem que ser mais preciso aqui.

"If a contribution edited an inner class while another deleted it" it = same inner class?

"sufficiently similar" é vago. join é vago. 

nao faz muito sentido um cara remover uma classe e adicionar uma referencia para ela ao mesmo tempo...